crates/rpc/rpc/src/engine.rs

```diff
@@ -157,12 +157,14 @@ impl EngineApiServer for EngineApi {
     /// Caution: This should not return the `withdrawals` field
     async fn get_payload_v1(&self, payload_id: PayloadId) -> Result<ExecutionPayload> {
         let (tx, rx) = oneshot::channel();
-        self.delegate_request(EngineApiMessage::GetPayload(payload_id, tx), rx).await
+        self.delegate_request(EngineApiMessage::GetPayload(payload_id, tx), rx)
+            .await
+            .map(ExecutionPayloadEnvelope::into_v1_payload)
     }
 
     /// Handler for `engine_getPayloadV2`
     /// See also <https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md#engine_getpayloadv2>
-    async fn get_payload_v2(&self, payload_id: PayloadId) -> Result<ExecutionPayload> {
+    async fn get_payload_v2(&self, payload_id: PayloadId) -> Result<ExecutionPayloadEnvelope> {
```

Comment:
This is kinda confusing, having `ExecutionPayload` and `ExecutionPayloadEnvelope` when they represent versions v1 and v2  of the same type.

Should we rename it to `ExecutionPayloadV2` and `ExecutionPayloadV1` so it matches the spec?

Reply:
> This is kinda confusing, having ExecutionPayload and ExecutionPayloadEnvelope when they represent versions v1 and v2 

yeah this is a bit weird but `engine_getPayloadV2` should still return without withdrawals pre shanghai

the response object for v2 returns the payload as field `executionPayload` but V1 returns the payload directly, and the payload's withdrawal field depends on 
`V1`: never
`V2`: only after shanghai